fineract-provider/src/main/java/org/apache/fineract/portfolio/savings/domain/SavingsAccount.java

```diff
@@ -1076,16 +1076,19 @@ private void payWithdrawalFee(final BigDecimal transactionAmount, final LocalDat
         for (SavingsAccountCharge charge : this.charges()) {
 
             if (charge.isWithdrawalFee() && charge.isActive()) {
-                if (charge.getFreeWithdrawalCount() == null) {
-                    charge.setFreeWithdrawalCount(0);
-                }
 
-                if (charge.isEnableFreeWithdrawal()) {
-                    resetFreeChargeDaysCount(charge, transactionAmount, transactionDate, user);
+                if (paymentDetail.getPaymentType().getPaymentName().equals(charge.getCharge().getPaymentType().getPaymentName())) {
```

Comment:
The logic of this PR is to allow withdrawal transactions by checking whether the payment type assigned to the charge is equal to the payment Type of the withdrawal transaction.

Reply:
condition 1: if the charge has both condition of specific paymentType and discountType (FINERACT-1374)
condition 2 : charge is assigned to specific payment type but normal charge(no discount Type)
condition 3:  discount charge but no condition of paymentType.
condition 4 : normal withdraw charges